core/src/style/material.cpp

```diff
@@ -0,0 +1,286 @@
+#include "material.h"
+
+#include <iostream>
+
+Material::Material():m_name("material"),
+m_emission(0.0), m_emission_texture(nullptr), m_emission_texture_scale(1.), m_emission_texture_mapping(MappingType::UV),
+m_ambient(1.0), m_ambient_texture(nullptr), m_ambient_texture_scale(1.), m_ambient_texture_mapping(MappingType::UV),
+m_diffuse(0.8), m_diffuse_texture(nullptr), m_diffuse_texture_scale(1.), m_diffuse_texture_mapping(MappingType::UV),
+m_specular(0.2), m_specular_texture(nullptr), m_specular_texture_scale(1.), m_specular_texture_mapping(MappingType::UV),
+m_shininess(0.2),
+m_normal_texture(nullptr), m_normal_texture_scale(1.), m_normal_texture_mapping(MappingType::UV),m_normal_texture_amount(1.),
+m_bEmission(false),m_bAmbient(false),m_bDiffuse(true),m_bSpecular(false) {
+}
+
+void Material::setEmission(const glm::vec4 _emission){
+	m_emission = _emission;
+    m_emission_texture.reset();
+    setEmissionEnabled(true);
+}
+
+void Material::setEmission(const std::string &_file, MappingType _type, glm::vec3 _scale, glm::vec4 _amount){
+    std::shared_ptr<Texture> texture(new Texture(_file));
+    setEmission(texture, _type, _scale, _amount);
+}
+
+void Material::setEmission(std::shared_ptr<Texture> _texture, MappingType _type, glm::vec3 _scale, glm::vec4 _amount){
+    m_emission_texture = _texture;
+    m_emission_texture_mapping = _type;
+    m_emission_texture_scale = _scale;
+    m_emission = _amount;
+    setEmissionEnabled((bool)_texture);
+}
+
+void Material::setAmbient(const glm::vec4 _ambient){
+	m_ambient = _ambient;
+    m_ambient_texture.reset();
+    setAmbientEnabled(true);
+}
+
+void Material::setAmbient(const std::string &_file, MappingType _type, glm::vec3 _scale, glm::vec4 _amount){
+    std::shared_ptr<Texture> texture(new Texture(_file));
```

Comment:
Yeah I think it makes sense for textures in general, unless there's some subtlety of the label textures that breaks this model. I'd say this change to textures should go into master first and then we can update this PR to use it.